packages/story-editor/src/components/library/panes/shopping/shoppingPane.js

```diff
@@ -73,53 +68,72 @@ function ShoppingPane(props) {
   const speak = useLiveRegion('assertive');
   const [loaded, setLoaded] = useState(false);
   const [isLoading, setIsLoading] = useState(false);
+  const [canLoadMore, setCanLoadMore] = useState(false);
   const [searchTerm, setSearchTerm] = useState('');
+  const [page, setPage] = useState(1);
   const [orderby, setOrderby] = useState('date');
   const [order, setOrder] = useState('desc');
-
   const [isMenuFocused, setIsMenuFocused] = useState(false);
   const [products, setProducts] = useState([]);
   const {
     actions: { getProducts },
   } = useAPI();
 
-  const isShoppingEnabled =
-    'none' !== shoppingProvider && isShoppingIntegrationEnabled;
+  const { currentPageProducts, deleteElementById, currentPageNumber } =
+    useStory(({ actions, state: { currentPage, currentPageNumber } }) => ({
+      currentPageProducts: currentPage?.elements
+        ?.filter(({ type }) => type === ELEMENT_TYPES.PRODUCT)
+        .map(({ id, product }) => ({
+          elementId: id,
+          product,
+        })),
+      deleteElementById: actions.deleteElementById,
+      currentPageNumber,
+    }));
 
-  const { currentPageProducts } = useStory(({ state: { currentPage } }) => ({
-    currentPageProducts: currentPage?.elements
-      ?.filter(({ type }) => type === ELEMENT_TYPES.PRODUCT)
-      .map(({ id, product }) => ({
-        elementId: id,
-        product,
-      })),
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
   }));
 
   const searchProducts = useCallback(
-    async (value = '', sortBy, sortOrder) => {
+    async (value = '', _page = 1, sortBy, sortOrder) => {
       try {
         setIsLoading(true);
-        setProducts(await getProducts(value, sortBy, sortOrder));
+        const { products: _products, hasNextPage } = await getProducts(
+          value,
+          _page,
+          sortBy,
+          sortOrder
+        );
+        if (_page === 1) {
+          setProducts(_products);
+        } else {
+          setProducts([...products, ..._products]);
+        }
+        setCanLoadMore(hasNextPage);
       } catch (err) {
         trackError('search_products', err?.message);
         showSnackbar({ message: err.message });
         setProducts([]);
+        setCanLoadMore(false);
+        setPage(1);
       } finally {
         setIsLoading(false);
         setLoaded(true);
       }
     },
-    [getProducts, showSnackbar]
+    [getProducts, products, showSnackbar]
   );
 
   const searchProductsWithTelemetry = useCallback(
-    (value = '', sortBy, sortOrder) => {
-      searchProducts(value, sortBy, sortOrder);
+    (value = '', _page, sortBy, sortOrder) => {
+      searchProducts(value, _page, sortBy, sortOrder);
       trackEvent('search', {
         search_type: 'products',
         search_term: value,
         search_order: sortOrder,
         search_orderby: sortBy,
+        search_page: _page,
```

Comment:
Good idea but I don't think we're interested in tracking individual pages in infinite scroll for now.